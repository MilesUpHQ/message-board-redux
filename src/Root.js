import React from "react";
import { createStore, combineReducers } from "redux";
import { Provider } from "react-redux";
import commentReducer from "./reducers/comment";
import PrivateReducer from "./reducers/private_message";

export default ({ children, initialState = {} }) => {
  const rootReducer = combineReducers({
    comments: commentReducer,
    privateMessages: PrivateReducer
  });

  const store = createStore(rootReducer, initialState);
  return <Provider store={store}>{children}</Provider>;
};
