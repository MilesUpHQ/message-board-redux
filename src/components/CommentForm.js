import React from "react";
import { connect } from "react-redux";
import * as actions from "../actions";

class CommentForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      commentInput: ""
    };
  }

  handleChange = event => {
    this.setState({
      commentInput: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.addComment(this.state.commentInput);
    this.setState({
      commentInput: ""
    });
  };
  render() {
    return (
      <form>
        <textarea
          onChange={this.handleChange}
          type="text"
          className="form-control"
          value={this.state.commentInput}
          placeholder="comment"
        />
        <button onClick={this.handleSubmit} className="btn btn-primary">
          Add
        </button>
      </form>
    );
  }
}

export default connect(
  null,
  actions
)(CommentForm);

// const mapDispatchToProps = (dispatch, ownProps) => {
//   return {
//     addComment: comment => {
//       dispatch(actions.addComment(comment));
//     }
//   };
// };

// export default connect(
//   null,
//   mapDispatchToProps
// )(CommentForm);
