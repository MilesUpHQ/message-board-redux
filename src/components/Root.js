import React, { Component } from "react";
import { createStore, combineReducers, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import reduxPromise from "redux-promise";
import { BrowserRouter } from "react-router-dom";
import { postReducer } from "../reducers/post_reducer";

// const logger = store => next => action => {
//   const result = next(action);
//   console.log("Logger:", action.type, store.getState());
//   return result;
// };

const logger = store => {
  return next => {
    return action => {
      const result = next(action);
      console.log("Logger:", action.type, store.getState());
      return result;
    };
  };
};
const rootReducer = combineReducers({
  posts: postReducer
});

class Root extends Component {
  render() {
    const initialState = this.props.initialState ? this.props.initialState : {};
    const store = createStore(
      rootReducer,
      initialState,
      applyMiddleware(logger, reduxPromise)
    );

    return (
      <div>
        <Provider store={store}>
          <BrowserRouter>{this.props.children}</BrowserRouter>
        </Provider>
      </div>
    );
  }
}

export default Root;
