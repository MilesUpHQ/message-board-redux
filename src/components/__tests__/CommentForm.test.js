import React from "react";
import { mount } from "enzyme";
import CommentForm from "../comments/CommentForm";
import Root from "../../Root";
import renderer from "react-test-renderer";

let wrapper;

beforeEach(() => {
  wrapper = mount(
    <Root>
      <CommentForm />
    </Root>
  );
});

it("match snapshot", () => {
  wrapper = renderer.create(
    <Root>
      <CommentForm />
    </Root>
  );
  expect(wrapper.toJSON()).toMatchSnapshot();
});

it("renders a textarea", () => {
  expect(wrapper.find("textarea").length).toEqual(1);
});

it("renders a blue color button", () => {
  expect(wrapper.find("button.btn-primary").length).toEqual(1);
});

it("should update value onChange", () => {
  wrapper.find("textarea").simulate("change", {
    target: { value: "new" }
  });
  wrapper.update();
  expect(wrapper.find("textarea").prop("value")).toEqual("new");
  expect(wrapper.find("textarea").prop("placeholder")).toEqual("comment");
});

it("should empty input text after submit", () => {
  wrapper.find("textarea").simulate("change", {
    target: { value: "new" }
  });
  wrapper.update();
  wrapper.find("button").simulate("click");
  wrapper.update();
  expect(wrapper.find("textarea").prop("value")).toEqual("");
});

// it("using mock to find if function was called", () => {
//   const onChangeSpy = jest.fn();
//   const onChange = onChangeSpy;
//   let newWrapper = mount(
//     <Root>
//       <CommentForm onChange={onChange} />
//     </Root>
//   );
//   expect(onChangeSpy).not.toHaveBeenCalled();
//   newWrapper.find("textarea").simulate("change", {
//     target: { value: "new" }
//   });
//   expect(onChangeSpy).toHaveBeenCalled();
// });
