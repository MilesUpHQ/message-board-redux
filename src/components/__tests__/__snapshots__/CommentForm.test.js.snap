// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<form>
  <textarea
    className="form-control"
    onChange={[Function]}
    placeholder="comment"
    type="text"
    value=""
  />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;
